### ISSUE TITLE
[RasPi buster] aziot-edge_1.2.9-1_armhf.deb is not a valid archive

### ISSUE THREAD

**Issue by matsujirushi** [Azure/iotedge#6800]
## Current Behavior

aziot-edge cannot be installed.

## Steps to Reproduce

**Summary:**

```
sudo apt-get update
sudo apt-get install aziot-edge
```

**Detail:**

```
$ sudo apt-get install aziot-edge
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following package was automatically installed and is no longer required:
  python-colorzero
Use 'sudo apt autoremove' to remove it.
The following additional packages will be installed:
  aziot-identity-service
The following NEW packages will be installed:
  aziot-edge aziot-identity-service
0 upgraded, 2 newly installed, 0 to remove and 240 not upgraded.
Need to get 3,524 kB of archives.
After this operation, 35.5 MB of additional disk space will be used.
Do you want to continue? [Y/n]
Get:1 https://packages.microsoft.com/debian/10/prod buster/main armhf aziot-identity-service armhf 1.2.6-1 [3,098 kB]
Get:2 https://packages.microsoft.com/debian/10/prod buster/main armhf aziot-edge armhf 1.2.9-1 [426 kB]
Fetched 3,524 kB in 2s (2,118 kB/s)
E: Invalid archive member header
E: Prior errors apply to /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb
debconf: apt-extracttemplates failed: No such file or directory
Selecting previously unselected package aziot-identity-service.
(Reading database ... 154109 files and directories currently installed.)
Preparing to unpack .../aziot-identity-service_1.2.6-1_armhf.deb ...
Unpacking aziot-identity-service (1.2.6-1) ...
Selecting previously unselected package aziot-edge.
Preparing to unpack .../aziot-edge_1.2.9-1_armhf.deb ...
Unpacking aziot-edge (1.2.9-1) ...
dpkg-deb (subprocess): cannot copy archive member from '/var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb' to decompressor pipe: unexpected end of file or stream
dpkg-deb (subprocess): decompressing archive member: lzma error: unexpected end of input
dpkg-deb: error: <decompress> subprocess returned error exit status 2
dpkg: error processing archive /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb (--unpack):
 cannot copy extracted data for './usr/bin/iotedge' to '/usr/bin/iotedge.dpkg-new': unexpected end of file or stream
Errors were encountered while processing:
 /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

## Context (Environment)

Raspberry Pi OS buster 2022-09-22

## Additional Information

https://packages.microsoft.com/debian/10/prod/pool/main/a/aziot-edge/

Is aziot-edge_1.2.9-1_armhf.deb corrupted?

```
$ ar vx /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb
x - debian-binary
x - control.tar.gz
x - data.tar.xz
ar: /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb is not a valid archive
```

**Comment by arsing** [comment]
It does seem to be busted:

```
$ apt install aziot-identity-service aziot-edge

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  aziot-edge aziot-identity-service
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 3524 kB of archives.
After this operation, 35.5 MB of additional disk space will be used.
Get:1 https://packages.microsoft.com/debian/10/prod buster/main armhf aziot-identity-service armhf 1.2.6-1 [3098 kB]
Get:2 https://packages.microsoft.com/debian/10/prod buster/main armhf aziot-edge armhf 1.2.9-1 [426 kB]
Fetched 3524 kB in 1s (4117 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package aziot-identity-service.
(Reading database ... 7347 files and directories currently installed.)
Preparing to unpack .../aziot-identity-service_1.2.6-1_armhf.deb ...
Unpacking aziot-identity-service (1.2.6-1) ...
Selecting previously unselected package aziot-edge.
Preparing to unpack .../aziot-edge_1.2.9-1_armhf.deb ...
Unpacking aziot-edge (1.2.9-1) ...
dpkg-deb (subprocess): cannot copy archive member from '/var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb' to decompressor pipe: unexpected end of file or stream
dpkg-deb (subprocess): decompressing archive member: lzma error: unexpected end of input
dpkg-deb: error: <decompress> subprocess returned error exit status 2
dpkg: error processing archive /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb (--unpack):
 cannot copy extracted data for './usr/bin/iotedge' to '/usr/bin/iotedge.dpkg-new': unexpected end of file or stream
Errors were encountered while processing:
 /var/cache/apt/archives/aziot-edge_1.2.9-1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

But perhaps the bigger question is why it's only v1.2 even though we've released lots of new releases since then.

@damonbarry @yophilav @micahl Do you know why the Debian 10 packages on packages.microsoft.com are so old?

**Comment by micahl** [comment]
@arsing we originally intended to add Debian 10 and published a version before we learned of an issue on Buster that affected arm32 and decided against going further.  Dunno why that package would be broken.  Given that it is we should consider unpublishing it rather than leaving it up.

@matsujirushi see the _Known issue: Debian 10 (Buster) on ARMv7_ described in the [release notes for 1.3](https://github.com/Azure/azure-iotedge/releases/tag/1.3.0).  The same issue applies to 1.4 (which we would recommend you use instead of 1.2 or 1.3).

**Comment by micahl** [comment]
@matsujirushi is Debian 11 (Bullseye) an option for you?

**Comment by matsujirushi** [comment]
Hi @micahl ,
Thanks for your message.

The manufacturer of my Compute Module 4 device does not provide bullseye images.
I am asking the manufacturer if they can provide the bullseye image.

Therefore, I want to run it with buster at the moment.

**Comment by micahl** [comment]
@matsujirushi I see, thanks.  We do make Debian 10 packages available as a convenience for our 1.4 releases.  You can find the latest ones [here with the 1.4.3](https://github.com/Azure/azure-iotedge/releases/tag/1.4.3).  Please use those and be sure to apply the workaround.
